### PR TITLE
dude.go first draft

### PR DESCRIPTION
**Pull request by mrcsptr**
I have implemented a switch to build three grades depending on the position that will be occupied, so that later we just have to pick the right grade.

### CODE REVIEW

File: dude.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+
+)
+// Dude a type containing the name of the dude and his collection of results
```
Comment (moxar) [inline dude.go]: Dude is a type holding it's caracteristics.

Your documentation is too specific, and may have to change if the implementation grow.

File: dude.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+type Dude struct {
```
Comment (moxar) [inline dude.go]: go fmt .

File: dude.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ Name string
+ Results []DudeResult
+}
+
+// DudeResult contains one result, defining the position occupied then by the player, and the score obtained.
```
Comment (moxar) [inline dude.go]: It does not contains one result, it represents one result, being the pair of a position and score.

File: dude.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+// Definition of the values for a win (V), tie (T), and a loss (L) (unused for now).
+const(
+    W = 3
+    T = 2
+    L = 1
+)
+// Sum produces the current grades for the player, depending on the position that will be occupied: Attack (A), Middle (M), and Defense (D).
+func (du Dude) Sum() (int, int, int){
```
Comment (moxar) [inline dude.go]: the receiver should be `d` not `du` according to golang's conventions.

Moreover, this function would get messy with more than 3 positions. You need a generic method that takes a string as input and calculates one score. This method is not exported. Let's call it `score(pos string)`. Then, you need 3 specific and exported methods: ScoreA, ScoreM, ScoreD that take no input, and internaly call the score() method.

**Comment by mrcsptr** [comment]
I've been pretentious by thinking i would end parsing tonight. I lack some info to go on: I'm not familiar with the []riley.Dude, does that mean riley is a collection of Dude ? I also need to know what format do I use for the dude files. In the mean time, i started the parsing by copying what we did in config.go.

File: dude.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	A := d.Score("A")
+	return A
+}
+
+func (d Dude) ScoreM() int {
+	M := d.Score("M")
+	return M
+}
+
+func (d Dude) ScoreA() int {
```
Comment (moxar) [inline dude.go]: This won't compile, you have two `func (d Dude) ScoreA() int` defined.

File: dude.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	D := d.Score("D")
+	return D
+}
+
+// Score builds grade according to the position.
+func (d Dude) Score(p string) int {
```
Comment (moxar) [inline dude.go]: Do not expose this method.

File: dude.go [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+	for _, s := range d.Results {
+
+		if s.Position == p {
+			v += s.Score
+		}
+	}
+	return v
+}
+
+// String implements the fmt.Stringer interface. It displays the name of the dude, it's current grades, and the number of fights he did.
```
Comment (moxar) [inline dude.go]: Actualy, it returns the json representation of a dude. If you add some attributes to the dude type, they will be returned to.
This function does not display anything, it returns a string.

**Comment by moxar** [comment]
The content of a dude file is defined in the specification.

Reminder here:



```
# johny.txt
# this line is a comment. It show the history of matches for Johny.
A V # A for attack, V for victory
M D # M for middle, D for defeat
D T # D for defense, T for tie
```

**Comment by mrcsptr** [comment]
up.

File: data/filesystem/dude.go [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+func AllDudes(location string) ([]riley.Dude, error) {
+	// ... load dudes.
+	files, err := ioutil.ReadDir(location)
+	if err != nil {
+		return nil , err
+	}
+var d []Dude
+	for _, file := range files {
+        g :=GetDude(file.Name())
```
Comment (moxar) [inline data/filesystem/dude.go]: GetDude returns a dude, and an error.

File: data/filesystem/dude.go [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/moxar/riley"
+)
+
+func AllDudes(location string) ([]riley.Dude, error) {
+	// ... load dudes.
+	files, err := ioutil.ReadDir(location)
+	if err != nil {
+		return nil , err
+	}
+var d []Dude
```
Comment (moxar) [inline data/filesystem/dude.go]: rename this variable to `dudes`. Slice -> plural

File: data/filesystem/dude.go (hunk 8 above)
Comment (moxar) [reply data/filesystem/dude.go]: Also, g stands for nothing. Use the same letter as a usual dude receiver: d

File: data/filesystem/dude.go [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+	for _, file := range files {
+        g :=GetDude(file.Name())
+        d = append(d,g)
+	}
+}
+
+func GetDude(location string)(riley.Dude, error){
+    var d Dude
+    d.Name = location
```
Comment (moxar) [inline data/filesystem/dude.go]: The name of the dude located at `foo/bar/johny.txt` is `johny`, not `foo/bar/johny.txt`

File: data/filesystem/dude.go [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    results, err := os.Open(location)
+    if err!= nil {
+        return  results, err
```
Comment (moxar) [inline data/filesystem/dude.go]: results is not a riley.Dude.

```
return riley.Dude{}, err
```

**Pull request closed without merging**